src/export/bops/index.ts

```diff
@@ -0,0 +1,536 @@
+    return acc;
+  }, {});
+  if (Object.keys(nots).map(Boolean).length > 0) {
+    data.constraints = { ...data.constraints, ...nots };
+  }
+
+  // 4. work status
+  const workStatus = getWorkStatus(passport);
+  if (workStatus) data.work_status = workStatus;
+
+  // 5. keys
+  const bopsData = removeNilValues(
+    Object.entries(bopsDictionary).reduce((acc, [bopsField, planxField]) => {
+      acc[bopsField as keyof BOPSFullPayload] = passport.data?.[planxField];
+      return acc;
+    }, {} as Partial<BOPSFullPayload>)
+  );
+
+  // 6a. questions+answers array
+  const { proposal_details, feedback } = formatProposalDetails(flow, breadcrumbs);
+  data.proposal_details = proposal_details;
+
+  // 6b. optional feedback object
+  // we include feedback object if it contains at least 1 key/value pair
+  if (Object.keys(feedback).length > 0) {
+    data.feedback = feedback;
+  }
+
+  // 7. payment
+  const payment = passport?.data?.[GOV_PAY_PASSPORT_KEY] as GovUKPayment;
+  if (payment) {
+    data.payment_amount = toPence(payment.amount);
+    data.payment_reference = payment.payment_id;
+  }
+
+  // 8. flag data
+  try {
+    const result = getResultData(breadcrumbs, flow);
+    const { flag } = Object.values(result)[0];
+    data.result = removeNilValues({
+      flag: [flag.category, flag.text].join(" / "),
+      heading: flag.text,
+      description: flag.officerDescription,
+      override: passport?.data?.["application.resultOverride.reason"],
+    });
+  } catch (err) {
+    throw new Error(`Error setting flag result: ${err}`);
+  }
+
+  // 9. user role
+  const userRole = passport?.data?.["user.role"]?.toString();
+  if (userRole && USER_ROLES.includes(userRole)) data.user_role = userRole;
+
+  // 10. Works
+  const works: BOPSFullPayload["works"] = {};
+
+  const startedDate = parseDate(passport?.data?.["proposal.start.date"]);
+  if (startedDate) works.start_date = startedDate;
+
+  const completionDate = parseDate(
+    passport?.data?.["proposal.completion.date"]
+  );
+  if (completionDate) works.finish_date = completionDate;
+
+  if (!isEmpty(works)) data.works = works;
+
+  return {
+    ...data,
+    ...bopsData,
+    planx_debug_data: {
+      session_id: sessionId,
+      breadcrumbs,
+      passport,
+    },
+  };
+}
+
+const parseDate = (dateString: string | undefined): string | undefined => {
+  try {
+    if (dateString) {
+      return new Date(dateString).toISOString().split("T")[0];
+    }
+  } catch (err) {
+    throw new Error(`Error parsing date ${dateString}: ${err}`);
+  }
+};
+
+const getWorkStatus = (passport: Passport) => {
+  switch (passport?.data?.["application.type"]?.toString()) {
+    case "ldc.existing":
+      return "existing";
+    case "ldc.proposed":
+      return "proposed";
+  }
+};
+
+const extractFileDescriptionForPassportKey = (
+  passport: Passport["data"],
+  passportKey: string
+): string | undefined => {
+  try {
+    // XXX: check for .description or .reason as there might be either atm
```

Comment:
I was never really familiar with this syntax prior to working with John & Gunar but have picked it up since: I understand it as a sort of synonym to `TODO` comments, but for known caveats/tricky logic notes that aren't actually outstanding tasks. It's equally easy to globally search for and sets apart from "regular" comments.

If you don't want it polluting planx-core, we don't have to port it over from planx-new though :upside_down_face:

Reply:
I'd also vote for removing `XXX` - it's new to me from John/Gunar also and I've always read it to mean "Care needed here / this is an explanation of something unexpected" which is just sort of what comments would be anyway.